### PR TITLE
feat: add `strict` option to application schematic

### PR DESCRIPTION
**Pull request by micalevisk**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #597 _(only the first step)_

## What is the new behavior?

As of the first part to address the issue #597 this PR adds the `strict` option into the `application` schematic.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

I didn't request to work on the issue #597 but I already know what to do in https://github.com/nestjs/nest-cli after if this PR is merged, and finally closes that issue :)

### CODE REVIEW

File: src/lib/application/files/ts/tsconfig.json [hunk 1]
```diff
@@ -11,6 +11,11 @@
     "outDir": "./dist",
     "baseUrl": "./",
     "incremental": true,
-    "skipLibCheck": true
+    "skipLibCheck": true,
+    "strictNullChecks": <%= strict %>,
+    "noImplicitAny": <%= strict %>,
```
Comment (micalevisk) [inline src/lib/application/files/ts/tsconfig.json:16]: I'm not sure if I should keep all this options here, or create another tsconfig template for the strict mode.

**Review by jmcdo29 — APPROVED** [review]
LGTM. Seems like a pretty straightforward change to me

File: src/lib/application/files/ts/tsconfig.json (hunk 1 above)
Comment (jmcdo29) [reply src/lib/application/files/ts/tsconfig.json:16]: I like the idea of keeping these to help new devs know the options exist

**Comment by kamilmysliwiec** [comment]
LGTM thanks!

**Pull request merged**